pyproject.toml

```
@@ -3,62 +3,109 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 [build-system]
-requires = ["poetry-core>=1.0.7"]
-build-backend = "poetry.core.masonry.api"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
 
-[tool.poetry]
+[project]
 name = "antsibull-core"
 version = "2.0.0a1.post0"
 description = "Tools for building the Ansible Distribution"
-authors = ["Toshio Kuratomi <a.badger@gmail.com>", "Felix Fontein <felix@fontein.de>"]
-license = "GPL-3.0-or-later"
+license = "GPL-3.0-or-later AND BSD-2-Clause AND MIT AND PSF-2.0"
+license-files = {globs=["LICENSES/*.txt"]}
 readme = "README.md"
-repository = "https://github.com/ansible-community/antsibull-core"
-packages = [
-    { include = "antsibull_core", from="src" },
-    { include = "tests", format = "sdist" }
-]
 classifiers = [
     "Development Status :: 3 - Alpha",
     "Framework :: Ansible",
-    "Intended Audience :: Developers"
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Typing :: Typed",
+]
+requires-python = ">=3.9"
+dependencies = [
+    "aiofiles",
+    "aiohttp >= 3.0.0",
+    # major/minor was introduced here
+    "packaging >= 20.0",
+    "perky",
+    # pydantic v2 is a major rewrite
+    "pydantic >= 1.0.0, < 2.0.0",
+    "PyYAML",
+    "semantic_version",
+    # sh v2 has breaking changes.
+    # https://github.com/ansible-community/antsibull-core/issues/34
+    "sh >= 1.0.0, < 2.0.0",
+    # 0.5.0 introduces dict_config
+    "twiggy >= 0.5.0",
 ]
 
-[tool.poetry.urls]
+
+[[project.authors]]
+name = "Toshio Kuratomi"
+email = "a.badger@gmail.com"
+
+[[project.authors]]
+name = "Felix Fontein"
+email = "felix@fontein.de"
+
+[[project.maintainers]]
+name = "Felix Fontein"
+email = "felix@fontein.de"
+
+[[project.maintainers]]
+name = "Maxwell G"
+email = "maxwell@gtmx.me"
+
+[project.urls]
+"Source code" = "https://github.com/ansible-community/antsibull-core"
 "Code of Conduct" = "https://docs.ansible.com/ansible/latest/community/code_of_conduct.html"
 "Bug tracker" = "https://github.com/ansible-community/antsibull-core/issues"
 
-[tool.poetry.dependencies]
-python = "^3.9.0"
-aiofiles = "*"
-aiohttp = ">= 3.0.0"
-# major/minor was introduced here
-packaging = ">= 20.0"
-perky = "*"
-# pydantic v2 is a major rewrite
-pydantic = ">= 1.0.0 < 2.0.0"
-PyYAML = "*"
-semantic_version = "*"
-sh = ">= 1.0.0 < 2.0.0"
-# 0.5.0 introduces dict_config
-twiggy = ">= 0.5.0"
+[project.optional-dependencies]
+codeqa = [
+    "flake8 >= 6.0.0",
+    "pylint ~= 2.15.7",
```

Why not
```suggestion
    "pylint >= 2.15.7",
```
? Or if you want to keep the old behavior:
```suggestion
    "pylint ~= 2.15",
```